File: leetcode_69/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        result = 0
        if(x<2):
            return x
        left, right = 1, x
        while left <= right:
            mid = int((left+right)/2)
            mid_squared = mid*mid
            if mid_squared == x:
                return mid
            elif mid_squared > x:
                right = mid-1
            else:
                left = mid+1
        return round(right)

File: leetcode_69/solution.c
int mySqrt(int x) {
    if (x<2)
    {
        return x;
    }
    long left = 1;
    long right = x;
    long midsquared = -1000;
    long mid = -1000;
    while(right>=left)
    {
        mid = (long) ((long)(right+left)/2);
        midsquared = (long) mid*mid ;
        if(midsquared == x)
        {
            return mid;
        }
        else if(midsquared>x)
        {
            right = mid-1;
        }
        else
        {
            left = mid+1;
        }
    }
    return (int) round(right);
}
